principles.rst

```diff
@@ -303,3 +309,124 @@ down GHC even when those features are not active. Yet this principle is importan
 
 From `#378`_, slightly generalized.
+
+
+
+GHC stability principles
+--------------------------
+
+The Haskell ecosystem has a built-in tension between stability and innovation.
+
+One the one hand, breaking changes impose heavy costs:
+
+* Users keep having to update their code
+* Library authors keep having to update their code
+* Updating to a later version of a library can in turn force taking on new dependencies.
+
+These difficulties add friction, discourage adoption, and make Haskell an unpredictable investment.
+
+On the other hand, GHC has always sought to be a laboratory for innovation.  If the stability guarantees are too onerous, we risk imposing a different set of costs:
+
+* Volunteers may get discouraged by the hoops they have to jump through to get a change agreed.
+* The language may become stuck in a local optimum, because moving to a better design would require breaking changes.
+* GHC may become encrusted with cruft and workarounds that exist to preserve existing, but undesirable, behaviour.
+
+We can't navigate this tension with simple all-or-nothing rules.  We have to take each case on its merits, aware of both sets of costs.  Rather than starting the debate from scratch each time, we can have some general rules, or principles, to guide us.  The `GHC base library proposal <https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/051-ghc-base-libraries.rst>`_ made a start in this direction; this section discusses GHC itself.
+
+Assumptions
+~~~~~~~~~~~~
+
+We assume:
+
+* That we have adopted the  `GHC base library proposal <https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/051-ghc-base-libraries.rst>`_, which establishes the ``ghc-experimental`` package.
+* That we have adopted some form of `GHC Proposal #601 <https://github.com/ghc-proposals/ghc-proposals/pull/601>`_, which allows us to designate extension flags as Experimental or Stable.  (Maybe other categories too, but at least these two.)  Moreover, we have actually categorised existing extensions, at least far enough to identify the Stable ones.  This is still a work in progress.
+
+Terminology
+~~~~~~~~~~~~~~~
+
+We define a **stable Haskell package** as follows. A stable Haskell package
+
+* Does not use Experimental extensions.
+* Does not use experimental features.  (Examples: ``INCOHERENT`` pragmas, use of the Javascript or Wasm back end, builds on non-tier-1 platforms.)
+* Does not use ``-Werror`` in its default build configuration.
+* Explicitly specifies a language edition (``Haskell98``, ``GHC2021``), in the source code or the build configuration.
+* Depends only on stable Haskell packages.
+
+Boot packages shipped with GHC will be designated as stable or not in the documentation:
+
+* ``base`` is stable, despite depending on unstable packages.
+* Any GHC-internal packages (e.g. ``ghc-internal``, ``ghc-prim``, ``ghc-bignum``) are not stable and may change API without notice and between minor releases.
```

Comment:
```suggestion
* Any GHC-internal packages (e.g. ``ghc-internal``, ``ghc-prim``, ``ghc-bignum``) are not stable and may change API without notice and between minor releases of GHC (the package version itself still follows the PVP).
```
Just a clarification. Is that correct?

Reply:
The agreement with CLC was that `ghc-internals` follows PVP.

Reply:
Yes, the "GHC's base libraries" document says that internal libraries follow PVP, and that as a consequence internal libraries may get a major version bump between minor GHC releases (https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/051-ghc-base-libraries.rst#versions-and-backports).

This does risk an awkward situation if we want to backport a bug fix that necessarily changes an API, because a major version bump might not be possible. We could potentially avoid that risk by bumping the supermajor version of internal libraries on each GHC release. But the same problem arises for `base`, as described in the linked document.